src/storage/sql_table.cpp

```diff
@@ -19,64 +43,330 @@ SqlTable::SqlTable(BlockStore *const store, const catalog::Schema &schema, const
   TERRIER_ASSERT(attr_sizes.size() == NUM_RESERVED_COLUMNS,
                  "attr_sizes should be initialized with NUM_RESERVED_COLUMNS elements.");
 
+  // First pass through to accumulate the counts of each attr_size
   for (const auto &column : schema.GetColumns()) {
     attr_sizes.push_back(column.GetAttrSize());
   }
 
   auto offsets = storage::StorageUtil::ComputeBaseAttributeOffsets(attr_sizes, NUM_RESERVED_COLUMNS);
 
-  ColumnMap col_oid_to_id;
-  // Build the map from Schema columns to underlying columns
+  ColumnMap col_map;
+  InverseColumnMap inv_col_map;
+
+  // Build the maps between Schema column OIDs and underlying column IDs
   for (const auto &column : schema.GetColumns()) {
     switch (column.GetAttrSize()) {
       case VARLEN_COLUMN:
-        col_oid_to_id[column.GetOid()] = col_id_t(offsets[0]++);
+        inv_col_map[col_id_t(offsets[0])] = column.GetOid();
+        col_map[column.GetOid()] = col_id_t(offsets[0]++);
         break;
       case 8:
-        col_oid_to_id[column.GetOid()] = col_id_t(offsets[1]++);
+        inv_col_map[col_id_t(offsets[1])] = column.GetOid();
+        col_map[column.GetOid()] = col_id_t(offsets[1]++);
         break;
       case 4:
-        col_oid_to_id[column.GetOid()] = col_id_t(offsets[2]++);
+        inv_col_map[col_id_t(offsets[2])] = column.GetOid();
+        col_map[column.GetOid()] = col_id_t(offsets[2]++);
         break;
       case 2:
-        col_oid_to_id[column.GetOid()] = col_id_t(offsets[3]++);
+        inv_col_map[col_id_t(offsets[3])] = column.GetOid();
+        col_map[column.GetOid()] = col_id_t(offsets[3]++);
         break;
       case 1:
-        col_oid_to_id[column.GetOid()] = col_id_t(offsets[4]++);
+        inv_col_map[col_id_t(offsets[4])] = column.GetOid();
+        col_map[column.GetOid()] = col_id_t(offsets[4]++);
         break;
       default:
         throw std::runtime_error("unexpected switch case value");
     }
   }
 
-  auto layout = storage::BlockLayout(attr_sizes);
-  table_ = {new DataTable(block_store_, layout, layout_version_t(0)), layout, col_oid_to_id};
+  // Populate the default value map
+  //
+  // clang's memory analysis gets confused by the iterator here.  This is tied
+  // to the same false positive identified inside of the loop and is caused.
+  // NOLINTNEXTLINE
+  for (const auto &column : schema.GetColumns()) {
+    auto col_oid = column.GetOid();
+    auto *default_value = column.GetDefault();
+    // Only populate the default values of the columns which are new and have a default value
+    if (default_value_map_.Find(col_oid) == default_value_map_.End()) {
+      uint8_t attr_size = column.GetAttrSize();
```

Comment:
Note that the results of `GetAttrSize()` of varlen attributes have the sign bit set. You will allocate a large chunk of memory.